components/treelist/selection/cells.md

```diff
@@ -0,0 +1,253 @@
+
+### Equals Comparison
+
+The items in `SelectedCells` are compared against the items in the TreeList data in order to determine which cells will be highlighted. The default framework behavior is to compare objects by reference. The data item references may not match when:
+
+* The `SelectedCells` are obtained from a different data source than the all TreeList items, for example, from a separate service.
+
+In such cases, the selected cells may not appear as expected. You have to [override the `Equals` method of the TreeList model class]({%slug treelist-state%}#basics) so that the items are compared by a unique identifier rather than by reference. When you override `Equals`, it is also recommended to override the [`GetHashCode`](https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode) method.
+
+## Cell Selection and Other TreeList Features
+
+The selection feature behavior may vary when the TreeList configuration combines cell selection and other TreeList features, such as editing, virtualization, paging, templates. In such cases you need to consider certain limitation or include some modications.
+
+### Selection and Editing Modes
+
+When you want to edit a TreeList item, the cell selection has the following behavior:
+
+* Cell selection is not supported with [`Incell` edit mode]({%slug treelist-editing-incell%}) due to the overlapping pointer events that trigger selection and editing. If both features are enabled, only the editing will work.
+* [`Inline` edit mode]({%slug treelist-editing-inline%}) and [`Popup` edit mode]({%slug treelist-editing-popup%}) integrate with cell selection without limitations.
+
+### Selection and Virtual Scrolling
+
+When the TreeList has [virtual scrolling]({%slug treelist-virtual-scrolling%}), the component is able to select a range of cells with **Shift** only if all rows in that range are currently rendered. Consider the following scenario:
+
+1. Select a cell.
+1. Scroll down, so that virtualization kicks in and the rendered rows are no longer the same.
+1. Select another cell with **Shift**.
+
+In this case, the range selection will start from the first row that is currently rendered. Compare with [Selection and paging](#selection-and-paging) below.
+
+### Selection and Paging
+
+The `SelectedCells` collection persists across paging.
+
+### Selection and Templates
+
+When using [TreeList templates]({%slug treelist-templates-overview%}) with cell selection:
+
+* If you are using a [TreeList column template]({%slug treelist-templates-column%}) and you have a clickable element in the template, wrap this element in a container with a `@onclick:stopPropagation` directive. You can check the knowledge base article on [how to prevent row selection when the user clicks another component in the Grid column template]({%slug grid-kb-row-selection-in-column-template%}). It is for the Grid component, but the concept is identical and it applies for both row and cell selection.
```

Comment:
```suggestion
* If you are using a [TreeList column template]({%slug treelist-templates-column%}) and you have a clickable element in the template, wrap this element in a container with an `@onclick:stopPropagation` directive. You can check the knowledge base article on [how to prevent row selection when the user clicks another component in the Grid column template]({%slug grid-kb-row-selection-in-column-template%}). It is for the Grid component, but the concept is identical and it applies to both row and cell selection.
```